lib/mongoid_auto_increment/incrementor.rb

```diff
@@ -58,18 +59,36 @@ def collection
       end
 
       def query
-        { "seq_name" => @sequence }
+        @scope.merge("seq_name" => @sequence)
       end
     end
 
     def initialize(options=nil)
     end
 
-    def inc(sequence, options)
+    def inc(sequence, options, record)
       collection = options[:collection] || "sequences"
       seed = options[:seed].to_i
       step = options[:step] || 1
-      Sequence.new(sequence, collection, seed, step).inc
+      scope = resolve_scope(record, options[:scope])
+
+      Sequence.new(sequence, collection, seed, step, scope).inc
+    end
+
+    private
+
+    def resolve_scope(record, scope)
+      Array(scope).each_with_object({}) do |scope_item, query|
```

Comment:
@jnfeinstein 
It looks too complicated.
Why do you need ```Array(scope)```?
As far as I understand from your code, ```scope``` can be either ```nil``` or ```Symbol```, so you don't need to iterate through Array(scope), because there is will be only one iteration.

Reply:
`scope: :company, :user`

I pulled this code from mongoid so the usage would the same.

Reply:
So it can take an array as an argument?
Great!
Can you also add some tests for it?

Reply:
Yes.  Added a spec for array with mixed relation and field.

Reply:
Right, sorry.  One second...